RecoCTPPS/TotemRPLocal/src/TotemTimingRecHitProducerAlgorithm.cc

```diff
@@ -10,65 +10,69 @@
 
-TotemTimingRecHitProducerAlgorithm::TotemTimingRecHitProducerAlgorithm(
-    const edm::ParameterSet &iConfig)
-    : sampicConversions_(iConfig.getParameter<std::string>("calibrationFile")),
-      baselinePoints_(iConfig.getParameter<int>("baselinePoints")),
-      saturationLimit_(iConfig.getParameter<double>("saturationLimit")),
-      cfdFraction_(iConfig.getParameter<double>("cfdFraction")),
-      smoothingPoints_(iConfig.getParameter<int>("smoothingPoints")),
-      lowPassFrequency_(iConfig.getParameter<double>("lowPassFrequency")),
-      hysteresis_(iConfig.getParameter<double>("hysteresis")) {}
-
-void TotemTimingRecHitProducerAlgorithm::build(
-    const CTPPSGeometry *geom, const edm::DetSetVector<TotemTimingDigi> &input,
-    edm::DetSetVector<TotemTimingRecHit> &output) {
-  for (const auto &vec : input) {
+TotemTimingRecHitProducerAlgorithm::TotemTimingRecHitProducerAlgorithm(const edm::ParameterSet& iConfig) :
+  sampicConversions_(iConfig.getParameter<bool>("mergeTimePeaks"),
+                     iConfig.getParameter<edm::FileInPath>("calibrationFile").fullPath()),
+  baselinePoints_  (iConfig.getParameter<int>   ("baselinePoints")),
+  saturationLimit_ (iConfig.getParameter<double>("saturationLimit")),
+  cfdFraction_     (iConfig.getParameter<double>("cfdFraction")),
+  smoothingPoints_ (iConfig.getParameter<int>   ("smoothingPoints")),
+  lowPassFrequency_(iConfig.getParameter<double>("lowPassFrequency")),
+  hysteresis_      (iConfig.getParameter<double>("hysteresis"))
+{}
+
+void
+TotemTimingRecHitProducerAlgorithm::build(const CTPPSGeometry* geom, const edm::DetSetVector<TotemTimingDigi>& input,
+                                          edm::DetSetVector<TotemTimingRecHit>& output)
+{
+  for (const auto& vec : input) {
     const TotemTimingDetId detid(vec.detId());
 
-    float x_pos = 0.0f, y_pos = 0.0f, z_pos = 0.0f, x_width = 0.0f, y_width = 0.0f, 
-          z_width = 0.0f;
+    float x_pos = 0., y_pos = 0., z_pos = 0.;
+    float x_width = 0., y_width = 0., z_width = 0.;
 
     // retrieve the geometry element associated to this DetID ( if present )
-    const DetGeomDesc *det = geom->getSensorNoThrow(detid);
+    const DetGeomDesc* det = geom->getSensorNoThrow(detid);
 
     if (det) {
-      x_pos = det->translation().x(), y_pos = det->translation().y();
+      x_pos = det->translation().x();
+      y_pos = det->translation().y();
       z_pos = det->parentZPosition(); // retrieve the plane position;
 
       x_width = 2.0 * det->params()[0], // parameters stand for half the size
       y_width = 2.0 * det->params()[1],
       z_width = 2.0 * det->params()[2];
-    } else
-      edm::LogWarning("TotemTimingRecHitProducerAlgorithm")
-          << "Failed to retrieve a sensor for " << detid;
+    }
+    else
+      edm::LogWarning("TotemTimingRecHitProducerAlgorithm") << "Failed to retrieve a sensor for " << detid;
```

Comment:
remind me please how often does this happen?
Normally, I'd expect that the unpacker produces only valid digis to be used in reco.
If that's the case, this should be an exception instead of a warning.

Reply:
Indeed, we can throw an exception instead.